yt/data_objects/static_output.py

```diff
@@ -1068,7 +1147,7 @@ def set_units(self):
                               omega_lambda=self.omega_lambda,
                               omega_radiation=self.omega_radiation,
                               use_dark_factor = use_dark_factor,
-                              w_0 = w_0, w_a = w_a)
+                              w_0 = w_0, w_a = w_a, unit_registry=self.unit_registry)
```

Comment:
This will likely have unintended consequences. Most of the cosmology calculations return values in the comoving frame but over multiple redshifts. The only way to make this possible was to equate the comoving and proper reference frames in the cosmology calculator's unit registry. I think the dataset and its cosmology calculator need to have distinct unit registries.

Reply:
OK - can you give me a test case where this would matter?

Reply:
It looks like you've already changed it, but PR #2230 adds a test for this.